.github/workflows/flake8.yml

```diff
@@ -21,7 +21,10 @@ jobs:
     - name: Install pre-commit
       run: |
         python -m pip install --upgrade pip wheel
-        pip install -r requirements-dev.txt
+        pip install ruff -r requirements-dev.txt
+
+    - name: Run ruff
+      run: ruff --format github .
```

Comment:
Why not run it via pre-commit?

Reply:
Will pre-commit show GitHub annotations to new committers who have not done `pre-commit install`?

Reply:
Oh...  Now I see.  Fixed.

Reply:
Please remove the GitHub Action job `flake8` from required and add `ruff` in its place.

Reply:
Only @omab can do such changes, I don't have permissions.